src/utils/identityKey.ts

```diff
@@ -2,6 +2,46 @@ import aesjs from "aes-js";
 import crypto from "crypto";
 
 class IdentityKey {
+  public static sanityCheck(identityKey: string): boolean {
+    let key = identityKey;
+
+    const separatorIndex = key.indexOf(';')
+    if (separatorIndex === -1) {
+      return false;
+    }
+
+    if (key.startsWith("identityNo:")) {
+      if (key.length === separatorIndex + 1) {
+        return true;
+      }
+      key = key.substring(separatorIndex + 1);
+    }
+
+    let result = true;
+
+    while (result) {
+      const separatorIndex = key.indexOf(';')
+      if (separatorIndex + 1 === key.length) {
+        break;
+      }
+      const colonIndex = key.indexOf(':');
+      if (colonIndex === -1) {
+        result = false;
+        break;
+      }
+
+      const cipher = Buffer.from(key.substring(colonIndex + 1, separatorIndex - 1), "base64");
+      if (cipher.length !== 32) {
+        result = false;
+        break;
+      }
+
+      key.substring(separatorIndex + 1);
```

Comment:
```suggestion
      key.substring(separatorIndex + 1);
```
this line seems to have no effect.

Reply:
Yes, I forgot to assign the key to the new value. Fixed now.